tools/payments/client.go

```diff
@@ -0,0 +1,185 @@
+package payments
+
+import (
+	"bytes"
+	"context"
+	"crypto/sha256"
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/brave-intl/bat-go/libs/httpsignature"
+	"github.com/google/uuid"
+	"github.com/redis/go-redis/v9"
+)
+
+const (
+	preparePrefix = "prepare-"
+	submitPrefix  = "authorize-"
+)
+
+var (
+	payout        = strconv.FormatInt(time.Now().Unix(), 10)
+	prepareStream = preparePrefix + payout
+	submitStream  = preparePrefix + payout
+
+	prepareConfigStream = preparePrefix + "-configure-" + payout
```

Comment:
Will this need to be a static stream name some its discoverable, something like `prepare-config`